client/src/components/Categories/CategoryCard.js

```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { Card, CardText, CardBody,CardTitle, Button } from 'reactstrap';
-function CategoryCard(props){       
+function CategoryCard(props){ 
+if(props.category.hideCategory!="on"){
```

Comment:
the hiding and showing of categories should not be client-side. Just update the api/categories to query by { hidden: false } - no need send items through the network, then hide them in the client (also not secure, as a user can simply open their developer tools and see the full response)